multihost_test/Bugzillas/test_subid_ranges_configurable.py

```diff
@@ -0,0 +1,294 @@
+""" Automation of Shadow-utils bugs """
+
+import pytest
+import subprocess
+import time
+import os
+from sssd.testlib.common.utils import SSHClient
+
+
+def exceute_cmd(multihost, command):
+    """ execute cmd in client machine """
+    cmd = multihost.client[0].run_command(command)
+    return cmd
+
+
+class TestSubid(object):
+    """
+    This is for Shadow bugs automation
+    """
+    def test_subid_nochange_nsswitch(self, multihost,
+                                     create_backup,
+                                     create_localuser):
+        """
+        :Title: support pluggable data sources for
+        subid ranges configurable via /etc/nsswitch.conf
+        :id: 6ff93666-00c0-11ec-9be4-845cf3eff344
+        :bugzilla: https://bugzilla.redhat.com/show_bug.cgi?id=1859252
+        :steps:
+            1. Define user range in /etc/subuid
+            2. Define group range in /etc/subgid
+            3. Create a process for user
+            4. Check newuidmap command is working
+            with no change in /etc/nsswitch.conf
+            5. Check newgidmap command is working
+            with no change in /etc/nsswitch.conf
+        :expectedresults:
+            1. Should succeed
+            2. Should succeed
+            3. Should succeed
+            4. Should succeed
+            5. Should succeed
+        """
+        multihost.client[0].run_command("echo 'local_anuj:165536:65536'"
+                                        " > /etc/subuid")
+        multihost.client[0].run_command("echo 'local_anuj:165536:65536' "
+                                        "> /etc/subgid")
+        multihost.client[0].run_command("echo 'unshare -U bash' "
+                                        "> /tmp/anuj.sh")
+        user = 'local_anuj'
+        ssh1 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password='Secret123')
+        ssh2 = SSHClient(multihost.client[0].sys_hostname,
+                         username=user, password='Secret123')
+        (results1, results2, results3) = ssh1.exec_command('sh /tmp/anuj.sh')
+        find_id = "ps -ef | grep bash | grep local_| tail -1"
```

Comment:
`local_` should be `$user`?

Reply:
yes but we no need to use full name  our user is local_anuj .

Reply:
This is bad coding style. If you defined a variable to hold 'username' then it's better to use it everywhere.

Reply:
i have changed it to grep local_anuj

Reply:
[root@client1 ~]# ps -ef | grep bash 
root        1103       1  0 08:12 ?        00:00:00 /bin/bash /usr/sbin/ksmtuned
root        2179    2177  0 08:13 pts/0    00:00:00 -bash
local_a+   10131   10105  0 08:30 ?        00:00:00 bash
root       10202    2179  0 08:30 pts/0    00:00:00 grep --color=auto bash
[root@client1 ~]# 

This was reason i was using "ps -ef | grep bash | grep local_| tail -1"

Reply:
See system is printing only:  local_a+ 10131 10105 0 08:30 ? 00:00:00 bash  not full user name . i have to again switch back to ps -ef | grep bash | grep local_| tail -1

Reply:
`ps axo user:20,pid,comm`?